addOns/automation/src/main/java/org/zaproxy/addon/automation/gui/AutomationPanel.java

```diff
@@ -223,26 +213,22 @@ public String getDescription() {
         return loadPlanButton;
     }
 
+    public void runPlan(AutomationPlan plan) {
+        new Thread(
+                        () -> {
+                            try {
+                                ext.runPlan(plan, true);
+                            } catch (AutomationJobException e1) {
+                                View.getSingleton().showWarningDialog(e1.getMessage());
```

Comment:
Again, check `hasView()` - basically this should be done anywhere in the code where we might not have the GUI. It was fine inside the display panel as that only ran in the GUI. Now its been moved it could be called when running in cmd or daemon modes.